src/internal/operations/backup.go

```diff
@@ -140,18 +141,19 @@ func (op *BackupOperation) Run(ctx context.Context) (err error) {
 		observe.Complete()
 	}()
 
-	ctx, flushMetrics := events.NewMetrics(ctx, logger.Writer{Ctx: ctx})
-	defer flushMetrics()
+	var runnable bool
 
-	//-----
-	// Precheck
-	//-----
-	err = Precheck(ctx, op.account, op.Selectors.PathService(), op.Selectors.DiscreteOwner)
-	if err != nil {
+	// IsBackupRunnable checks if the user has services enabled to run a backup.
+	// it also checks for conditions like mailbox full.
+	runnable, op.canMakeDeltaQueries, err = op.bp.IsBackupRunnable(ctx, op.Selectors.PathService(), op.ResourceOwner.ID())
+	if err != nil && !runnable {
```

Comment:
when `err != nil && runnable` this conditional will get skipped, which doesn't feel like the correct behavior.  I'd suggest splitting the error and runnable checks into two separate conditions.

Reply:
my bad. Placed `&&` instead of `||` .